GetStream/stream-chat-android
# Move create channel side effects to LLC

closes #3163 

### 🎯 Goal

Move create channel side effects to LLC

### 🛠 Implementation details
1. Cleanup duplicated `ChatClient::createChannel` methods
2. Move `createChannel` side effects to LLC
3. Change `createChannel` behavior - now it doesn't do optimistic UI update as it's impossible to decide whether the channel should or should not be included in `ChannelListView`
4. Introduce `OfflineErrorHandlerFactoriesProvider ` responsible for providing offline-related error handler factories
5. Fix create channel sync flow
6. Migrate tests

### 🧪 Testing
Try to create a named channel offline:
1. Click add channel button in `ChannelListHeader`
2. Click `Create a Group`
3. Choose Members
4. Choose a channel name
5. Turn on airplane mode
6. Click create channel
7. Go back to `ChannelListView`, channel shouldn't not be there
8. Turn off airplane mode
9. Observe that channel is added to the `ChannelListView`

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
